client/src/pages/BrowsePage.js

```diff
@@ -18,22 +17,37 @@ const Container = styled.div`
 `;
 
 export const BrowsePage = () => {
-  const { data, loading, error, doFetch } = useAsync(() => getData(all));
+  const [method, setMethod] = useState(getData);
+  const { data, loading, error, doFetch } = useAsync(() => method);
+  const [tag, setTag] = useState("");
 
   useEffect(() => {
     doFetch();
-  }, []);
+  }, [method]);
 
+  const handleChange = (event) => {
+    setTag(event.target.value);
+  };
+
+  const handleSubmit = (event) => {
+    event.preventDefault();
+    setMethod(getProjectByTag(tag));
+  };
   return (
     <>
       <Header title={"Browse Projects"} />
       <Container>
-        <Searchbar />
+        <Searchbar
+          onSubmit={handleSubmit}
+          name="tag"
+          value={tag}
+          onChange={handleChange}
+        />
```

Comment:
In this case, it's enough to listen for `onSubmit`.
`value` and `onChange` could be in the `Searchbar` commit (move state) and `name` is not used.
This will reduce the complexity of `BrowserPage`.

```suggestion
        <Searchbar onSubmit={handleSubmit} />
```